### DISCUSSION TITLE
Identify frame origin when using multiple streams

### DISCUSSION THREAD

**Discussion by koenvboheemen** [miguelaeh/pipeless#150] [Q&A]
Hello and thank you for the awesome work this very nice project.

Using the docs I've been able to build a processing pipeline which I'd now like to use to process multiple streams.
As all streams must be processed similarly, Pipeless is very usefull for this purpose.
However, I am looking for a way to correlate processing results to the input stream the frame originates from.
In other words: If I count 5 objects in a frame, I need to know from which of my multiple input streams this frame originates to be able to use my detection result.

Would you please be so kind as to advise if and how I can find out from which input stream a frame originates?

**Comment by miguelaeh** [comment]
Hi @koenvboheemen ,
If I remember correctly, there should be a `stream_uuid` field on the frame data. That should match the uuid of the input stream.
The only objection with that is that you cannot know it before hand, as the stream uuid is dynamically generated when the stream is added.

**Reply by koenvboheemen** [reply]
Thank you @miguelaeh for your swift reply!

Upon inspecting the frame data object I found it to contain the fields as described in [frame-data-fields](https://www.pipeless.ai/docs/docs/v1/getting-started#frame-data-fields). Additionally, it contains a `dts` field which appears to be `0` all the time and a `pipeline_id` field which indicates which pipeline the data belongs to. I was not able to find a field containing info related to the origin of the frame.
Excuse me if I have overlooked the field(s) you are referring to.

It would be amazing if a user could, upon adding a new stream, add a flag or something similar which would for example be accessible as a separate field in frame data or maybe via the `user_data` field in the frame data.

**Reply by miguelaeh** [reply]
Hi @koenvboheemen ,
That suggestion makes sense, adding a flag to name streams beforhand could help on what you need.
BTW doesn't the `pipeline_id` field match the id when you list the streams with the CLI?

**Reply by koenvboheemen** [reply]
Thank you for your reply!

I was able to work around the problem by programming the cameras providing my streams to add a text-stamp in the corner of the image before sending the data to Pipeless. In pre-processing I read the text-stamp so that I know which camera my data originates from.

Would still be cool to have this on the feature-request list :)

And yes, the CLI tells me which streams are being processed by which pipeline, that part works as it should.

**Reply by miguelaeh** [reply]
Glad to hear that!
